lms/djangoapps/ccx/overrides.py

```diff
@@ -130,23 +137,33 @@ def override_field_for_ccx(ccx, block, name, value):
     field = block.fields[name]
     value_json = field.to_json(value)
     serialized_value = json.dumps(value_json)
-    try:
-        override = CcxFieldOverride.objects.create(
-            ccx=ccx,
-            location=block.location,
-            field=name,
-            value=serialized_value
-        )
-    except IntegrityError:
-        transaction.commit()
-        override = CcxFieldOverride.objects.get(
-            ccx=ccx,
-            location=block.location,
-            field=name
-        )
+    has_changes = False
+
+    override = get_override_for_ccx(ccx, block, name + "_instance")
+    if override:
+        has_changes = serialized_value != override.value
+
+    if not override:
+        try:
+            override = CcxFieldOverride.objects.create(
+                ccx=ccx,
+                location=block.location,
+                field=name,
+                value=serialized_value
+            )
+            _get_overrides_for_ccx(ccx).setdefault(block.location, {})[name + "_id"] = override.id
+        except IntegrityError:
+            transaction.commit()
```

Comment:
@cewing would you know anything more about this manual transaction commit when saving a ccx schedule? I remember we had some issues with spurious integrity errors -- was this part of a workaround? 

This was originally part of b3da2a5.